fixtures/js-stubs/metricRule.ts

```diff
@@ -1,10 +1,16 @@
+import {Dataset} from 'sentry/views/alerts/rules/metric/types';
+import {CombinedAlertType, CombinedMetricIssueAlerts} from 'sentry/views/alerts/types';
+
 import {IncidentTrigger} from './incidentTrigger';
 
-export function MetricRule(params = {}) {
+export function MetricRule<
+  T extends CombinedMetricIssueAlerts & {type: CombinedAlertType.METRIC}
+>(params: Partial<T> = {}): CombinedMetricIssueAlerts {
```

Comment:
I'm not sure this makes sense to do just to get the alert list tests to pass

Reply:
it's a `MetricRule` generator, and the type `CombinedMetricIssueAlerts` allows for two types of rules, so imo it's proper to scope things down, the return value doesn't need to be either-or

Reply:
i mean having the `type: CombinedAlertType.` since this was made to return a metric alert object